x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/cohere/embeddings/CohereEmbeddingsTaskSettings.java

```diff
@@ -72,10 +75,27 @@ public static CohereEmbeddingsTaskSettings fromMap(Map<String, Object> map) {
         return new CohereEmbeddingsTaskSettings(inputType, truncation);
     }
 
+    private final InputType inputType;
+    private final CohereTruncation truncation;
+
     public CohereEmbeddingsTaskSettings(StreamInput in) throws IOException {
         this(in.readOptionalEnum(InputType.class), in.readOptionalEnum(CohereTruncation.class));
     }
 
+    public CohereEmbeddingsTaskSettings(@Nullable InputType inputType, @Nullable CohereTruncation truncation) {
+        validateInputType(inputType);
+        this.inputType = inputType;
+        this.truncation = truncation;
+    }
+
+    private static void validateInputType(InputType inputType) {
+        if (inputType == null) {
+            return;
+        }
+
+        assert VALID_INPUT_VALUES_LIST.contains(inputType) : invalidInputTypeMessage(inputType);
```

Comment:
We shouldn't allow storing the `UNSPECIFIED` value or reading it from the `task_settings`.

Reply:
This should be a programmatic error if `UNSPECIFIED` is passed to this class in the constructor. I'm happy to make it an exception if that's people's preference. From a quick search, seems like there are examples of assertions similar to this at least in the ml plugin 🤷‍♂️